pkg/backend/openshiftcluster.go

```diff
@@ -102,12 +102,31 @@ func (ocb *openShiftClusterBackend) handle(ctx context.Context, log *logrus.Entr
 		return err
 	}
 
-	hiveRestConfig, err := ocb.env.LiveConfig().HiveRestConfig(ctx, 1)
+	// Only attempt to access Hive if we are installing via Hive or adopting clusters
+	installViaHive, err := ocb.env.LiveConfig().InstallViaHive(ctx)
 	if err != nil {
-		log.Info(err) // TODO(hive): Update to fail once we have Hive everywhere in prod and dev
+		return err
+	}
+
+	adoptViaHive, err := ocb.env.LiveConfig().AdoptByHive(ctx)
+	if err != nil {
+		return err
+	}
+
+	var hr hive.ClusterManager
+	if installViaHive || adoptViaHive {
+		hiveShard := 1
+		hiveRestConfig, err := ocb.env.LiveConfig().HiveRestConfig(ctx, hiveShard)
+		if err != nil {
+			return fmt.Errorf("failed getting RESTConfig for Hive shard %d: %w", hiveShard, err)
+		}
+		hr, err = hive.NewFromConfig(log, ocb.env, hiveRestConfig)
```

Comment:
what does `hr` mean in this context? I can see h for hive, but `hr` doesn't mean anything to me